python/cuspatial/cuspatial/tests/binpreds/binpred_test_dispatch.py

```diff
@@ -0,0 +1,535 @@
+        """
+      x
+      |\\
+    --|-x
+    | |/|
+    | x |
+    |   |
+    x----
+    """,
+        Polygon([(0.5, 0.5), (0.5, 1.5), (1.0, 1.0)]),
+        point_polygon,
+    ),
+    "polygon-polygon-in-point-point": (
+        """
+    x----
+    |\\  |
+    | x |
+    |/  |
+    x----
+    """,
+        Polygon([(0.0, 0.0), (0.0, 1.0), (0.5, 0.5)]),
+        point_polygon,
+    ),
+    "polygon-polygon-contained": (
+        """
+    -----
+    |  x|
+    | /||
+    |x-x|
+    -----
+    """,
+        Polygon([(0.25, 0.25), (0.75, 0.75), (0.75, 0.25)]),
+        point_polygon,
+    ),
+    "polygon-polygon-same": (
+        """
+    x---x
+    |   |
+    |   |
+    |   |
+    x---x
+    """,
+        point_polygon,
+        point_polygon,
+    ),
+}
+
+point_point_dispatch_list = [
+    "point-point-disjoint",
+    "point-point-equal",
+]
+
+point_linestring_dispatch_list = [
+    "point-linestring-disjoint",
+    "point-linestring-point",
+    "point-linestring-edge",
+]
+
+point_polygon_dispatch_list = [
+    "point-polygon-disjoint",
+    "point-polygon-point",
+    "point-polygon-edge",
+    "point-polygon-in",
+]
+
+linestring_linestring_dispatch_list = [
+    "linestring-linestring-disjoint",
+    "linestring-linestring-same",
+    "linestring-linestring-touches",
+    "linestring-linestring-crosses",
+]
+
+linestring_polygon_dispatch_list = [
+    "linestring-polygon-disjoint",
+    "linestring-polygon-touch-point",
+    "linestring-polygon-touch-edge",
+    "linestring-polygon-overlap-edge",
+    "linestring-polygon-intersect-edge",
+    "linestring-polygon-intersect-inner-edge",
+    "linestring-polygon-point-interior",
+    "linestring-polygon-edge-interior",
+    "linestring-polygon-in",
+    "linestring-polygon-crosses",
+]
+
+polygon_polygon_dispatch_list = [
+    "polygon-polygon-disjoint",
+    "polygon-polygon-touch-point",
+    "polygon-polygon-touch-edge",
+    "polygon-polygon-overlap-edge",
+    "polygon-polygon-point-inside",
+    "polygon-polygon-point-outside",
+    "polygon-polygon-in-out-point",
+    "polygon-polygon-in-point-point",
+    "polygon-polygon-contained",
+    "polygon-polygon-same",
+]
+
+
+def object_dispatch(name_list):
+    """Generate a list of test cases for a given set of test names."""
+    for name in name_list:
+        yield (name, features[name][0], features[name][1], features[name][2])
+
+
+type_dispatch = {
+    # A dictionary of test cases for each geometry type combination.
+    # Still needs MultiPoint.
+    (Point, Point): object_dispatch(point_point_dispatch_list),
+    (Point, LineString): object_dispatch(point_linestring_dispatch_list),
+    (Point, Polygon): object_dispatch(point_polygon_dispatch_list),
+    (LineString, LineString): object_dispatch(
+        linestring_linestring_dispatch_list
+    ),
+    (LineString, Polygon): object_dispatch(linestring_polygon_dispatch_list),
+    (Polygon, Polygon): object_dispatch(polygon_polygon_dispatch_list),
+}
+
+
```

Comment:
I started out just parameterizing the cases in the fashion that you describe. Because each of the feature combinations is predefined, I needed to have a separate test with separate parameterizations for exclusive set of features (point, linestring), (point, polygon), etc. Because I wanted logging and reporting on the test results, it was much easier to move that into a single test, and a single list of features that are used in the test. The primary contribution of this PR is the hand-coded list in `features`.

Reply:
I remember the specific reason that I parameterize the tests according to the pattern in https://github.com/rapidsai/cuspatial/pull/1085/files#diff-7064c17cb647ac9ab6d1e6f5c4c1726f68fd5b20205dd85f1d34684d53fd3ea6R20-R35 instead of using the pattern that you described. The simple answer is because you cannot parameterize a pytest fixture, only native iterables. I am using the `@pytest.mark.parameterize` functionality, I've just moved the generation of the tests into the fixture instead of passing a list to the `@pytest.mark.parameterize` call.